pkg/kine/server/list.go

```diff
@@ -28,20 +26,17 @@ func (l *LimitedServer) list(ctx context.Context, r *etcdserverpb.RangeRequest)
 		return nil, fmt.Errorf("invalid range end length of 0")
 	}
 
-	prefix := string(append(r.RangeEnd[:len(r.RangeEnd)-1], r.RangeEnd[len(r.RangeEnd)-1]-1))
-	if !strings.HasSuffix(prefix, "/") {
-		prefix = prefix + "/"
-	}
-	start := string(bytes.TrimRight(r.Key, "\x00"))
+	start := r.Key
```

Comment:
IMHO these two variables can be removed altogether. It's not a lot of typing to write `r.Key` instead of `key` :-)